src/image-viewer.js

```diff
@@ -549,23 +584,33 @@ export class ImageViewer {
    * @return {!Promise}
    * @private
    */
-  onZoomRelease_(centerClientX, centerClientY, deltaX, deltaY, veloX, veloY) {
+  onTapZoomRelease_(centerClientX, centerClientY,
+    deltaX, deltaY, veloX, veloY) {
     let promise;
     if (veloX == 0 && veloY == 0) {
       promise = Promise.resolve();
     } else {
       promise = continueMotion(this.image_,
           deltaX, deltaY, veloX, veloY,
           (x, y) => {
-            this.onZoomInc_(centerClientX, centerClientY, x, y);
+            this.onTapZoom_(centerClientX, centerClientY, x, y);
             return true;
           }).thenAlways();
     }
+    return promise.then(() => {
+      this.onZoomRelease_();
+    });
+  }
 
+  /**
+   * Performs actions after the gesture that was performing zooming has been
+   * released.
+   * @return {!Promise}
+   * @private
+   */
```

Comment:
Basically refactored out this shared portion so that `DoubleTap` zoom and `Pinch` zoom can use this without unnecessary variables.